spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteManifestsSparkAction.java

```diff
@@ -160,6 +172,32 @@ public RewriteManifestsSparkAction stagingLocation(String newStagingLocation) {
     return this;
   }
 
+  @Override
+  public RewriteManifestsSparkAction clusterBy(List<String> clusteringColumns) {
+    // Collect set of allowable partition columns to cluster on
+    Set<String> availablePartitionNames =
+        spec.fields().stream().map(PartitionField::name).collect(Collectors.toSet());
+
+    // Check if these partition fields are included in the spec
+    Preconditions.checkArgument(
```

Comment:
Do we care about case sensitivity here?

Reply:
I believe partition columns are saved verbatim. Are `FOO` and `foo` considered the same or different columns? We could desensitize this operation to case